geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/Transformers.scala

```diff
@@ -540,6 +545,25 @@ class GeometryFunctionFactory extends TransformerFunctionFactory {
         throw new IllegalArgumentException(s"Invalid geometrycollection conversion argument: ${args.toList}")
     }
   }
+
+  val reprojectParserFn = new TransformerFn {
+
+    private val cache = new SoftThreadLocalCache[String, MathTransform]
+
+    override val names: Seq[String] = Seq("reproject")
+
+    override def eval(args: Array[Any])(implicit ctx: EvaluationContext): Any = {
+      import org.locationtech.geomesa.utils.geotools.CRS_EPSG_4326
+
+      val geom = args(0).asInstanceOf[Geometry]
+      val epsg = args(1).asInstanceOf[String]
+      val lenient = if (args.length > 2) { java.lang.Boolean.parseBoolean(args(2).toString) } else { true }
+      // assuming here that transforms are not thread safe - can't find anything to say if they are or not
```

Comment:
Might be worth asking the GeoTools community.

Reply:
According to Jody they are thread safe, so I'll update it: http://osgeo-org.1560.x6.nabble.com/Are-MathTransform-implementations-thread-safe-td5129781.html